### PR TITLE
docs/esp32: Add UART to quickref.

### PR DESCRIPTION
**Pull request by mcauser**
Screen Shot 2021-04-30 at 2 42 50 pm

### CODE REVIEW

File: docs/esp32/quickref.rst [hunk 1]
```diff
@@ -171,6 +171,30 @@ Notes:
...
+ESP32 variant and board, these pins may conflict with onboard peripherals.
+
+Any GPIO can be used for hardware UARTs using the GPIO matrix, so to avoid
+conflicts simply provide ``tx`` and ``rx`` pins when constructing.
+
+UART0 (GPIO1=TX, GPIO3=RX) is dedicated to the REPL.
+
+UART1 (GPIO10=TX, GPIO9=RX) default pins may conflict if your board uses PSRAM.
+
+UART2 (GPIO17=TX, GPIO16=RX) default pins may conflict with embedded flash.
```
Comment (dpgeorge) [inline docs/esp32/quickref.rst:196]: might be better to make a table with these defaults, to match the SPI and I2C sections

File: docs/esp32/quickref.rst [hunk 2]
```diff
@@ -171,6 +171,30 @@ Notes:
...
+
+See :ref:`machine.UART <machine.UART>`. ::
+
+    from machine import UART
+
+    uart1 = UART(1, baudrate=9600, tx=33, rx=32)
+    uart1.write('hello')  # write 5 bytes
+    uart1.read(5)         # read up to 5 bytes
+
+The ESP32 has three hardware UARTs. UART0, UART1 and UART2.
```
Comment (dpgeorge) [inline docs/esp32/quickref.rst:185]: IMO would read better with a colon: "The ESP32 has three hardware UARTs: UART0, UART1 and UART2."

**Comment by mcauser** [comment]
Screen Shot 2021-05-05 at 10 18 13 am

@dpgeorge Updated.

File: docs/esp32/quickref.rst (hunk 2 above)
Comment (mcauser) [reply docs/esp32/quickref.rst]: :+1:

File: docs/esp32/quickref.rst (hunk 1 above)
Comment (mcauser) [reply docs/esp32/quickref.rst]: Implemented.

**Comment by dpgeorge** [comment]
Thanks for updating!  Merged in b98197f950b572aff2a34934e64d094e5cea6cd5

**Pull request closed without merging**